Midstone/ComponentFramework/Explosion.cpp

```diff
@@ -0,0 +1,173 @@
+#include "Explosion.h"
+
+
+bool Explosion::OnCreate(Camera* cam_, Shader* loadVerts, Mesh* mesh_, Model* model_)
+{
+	mesh = mesh_;
+	cam = cam_;
+
+	//create the buffer to hold the vertex positions between the vert stage and compute "stage"
+	glGenBuffers(1, &posBuffer);
+	glBindBuffer(GL_SHADER_STORAGE_BUFFER, posBuffer);
+	glBufferData(GL_SHADER_STORAGE_BUFFER, sizeof(Vec3) * 10000, NULL, GL_DYNAMIC_COPY); //null canse the loadVertsToBuffer shader will handle it
```

Comment:
If the 10000 is same as line 20 maybe turn it into constant

Reply:
yeah I should probably do that